src/multipa/main.py

```diff
@@ -471,6 +488,11 @@ def main_cli():
     full_valid_data = full_valid_data.cast_column("audio", Audio(sampling_rate=16_000))
     print("Sampling rate adjustment done")
 
+    # Final filter to check that the samples are valid
+    print("Filtering the dataset to remove invalid samples...")
+    full_train_data = full_train_data.filter(is_valid_sample)
+    full_valid_data = full_valid_data.filter(is_valid_sample)
```

Comment:
I think this might help us make best use of Unity resources for multiprocessing. Can you please try it out before committing? 

```suggestion
    full_train_data = full_train_data.filter(is_valid_sample, args.num_proc)
    full_valid_data = full_valid_data.filter(is_valid_sample, args.num_proc)
```